src/requestHandler.js

```diff
@@ -0,0 +1,47 @@
+const { parse: parseQuery } = require('querystring');
+const { URL } = require('url');
+const router = require('./router');
+
+function internalServerError(res, err) {
+  const errMess = { message: 'Internal error occurred' };
+  if (process.env.NODE_ENV === 'development') errMess.Error = err;
+
+  res.setHeader('Content-Type', 'application/json');
+  res.statusCode = 500;
+  res.write(JSON.stringify(errMess));
+  res.end();
+}
+
+module.exports = async (request, response) => {
+  try {
+    const { url } = request;
+    const parsedUrl = new URL(url, process.env.ORIGIN);
+    const queryParams = parseQuery(parsedUrl.search.substr(1));
+
+    let body = [];
+
+    request
+      .on('error', err => {
+        throw new Error(err);
```

Comment:
This code throws a caught error which makes it an uncaught error. Why throwing here?

Reply:
I have `catch` over here in 44 - 47 lines. Is it not work right?

Reply:
Right. This will throw an error inside `request.on()` function, which has no `try/catch` inside.
This is the main purpose of the `error` event. Whenever an error occurs in the request it will be emitted as `error` and should be handled there. At this point it cannot be handled in any way, so just logging this error would be a good solution.